packages/kit/src/runtime/app/forms.js

```diff
@@ -28,7 +28,7 @@ export function enhance(form, submit = () => {}) {
 			await invalidateAll();
 		}
 
-		if (location.origin + location.pathname === action.split('?')[0]) {
+		if (location.origin + location.pathname === action.origin + action.pathname) {
```

Comment:
What's the reason for having this `if` statement?
Would removing it fix #6695 ?

Reply:
It would fix it, yes, but the if check is there deliberately. I'll share more details on the issue.